io/disk/multipath_test.py

```diff
@@ -151,46 +153,150 @@ def test(self):
+        self.mpath_svc.restart()
+        wait.wait_for(self.mpath_svc.status, timeout=10)
+        if err_paths:
+            self.fail("following paths fails in all paths : %s" % err_paths)
+
+    def test_removing_all_paths(self):
+        '''
+        Removing and adding back all paths Dynamically using multipathd
+        '''
+        err_paths = []
+        self.log.info("Removing and adding back all paths dynamically")
+        for dic_path in self.mpath_list:
+            for path in dic_path["paths"]:
+                if multipath.remove_path(path) is False:
+                    self.log.info("couldn't remove in remove path: %s", path)
+                    err_paths.append(path)
+
+            time.sleep(self.op_long_sleep_time)
+            for path in dic_path["paths"]:
+                if multipath.add_path(path) is False:
+                    self.log.info("couldn't add back in all path: %s", path)
+                    err_paths.append(path)
             self.mpath_svc.restart()
             wait.wait_for(self.mpath_svc.status, timeout=10)
+        if err_paths:
+            self.fail("failed paths in remove indvdl paths: %s" % err_paths)
 
-        # Print errors
-        if msg:
-            self.fail("Some tests failed. Find details below:\n%s" % msg)
+    def test_suspend_resume_individual_mpath(self):
+        '''
+        suspending the mpathX and Resume it Back
+        '''
+        err_mpaths = []
+        self.log.info("Suspending mpaths and resuming them Back")
+        for dic_mpath in self.mpath_list:
+            if multipath.suspend_mpath(dic_mpath["name"]) is False:
+                self.log.info("couldn't suspend : %s", dic_mpath['name'])
+                err_mpaths.append(dic_mpath["name"])
+
+            time.sleep(self.op_shot_sleep_time)
+            if multipath.resume_mpath(dic_mpath["name"]) is False:
+                self.log.info("couldn't resume: %s", dic_mpath["name"])
+                err_mpaths.append(dic_mpath["name"])
+        if err_mpaths:
+            self.fail("error mpaths in suspnd indvdl mpaths: %s" % err_mpaths)
+
+    def test_suspend_resume_all_mpath(self):
+        '''
+        suspending all the mpathX and then Resume it Back at once
+        '''
+        err_mpaths = []
+        self.log.info("Suspending mpaths and resuming them Back")
+        for dic_mpath in self.mpath_list:
+            if multipath.suspend_mpath(dic_mpath["name"]) is False:
+                self.log.info("couldn't suspend %s", dic_mpath["name"])
+                err_mpaths.append(dic_mpath['name'])
+
+        time.sleep(self.op_long_sleep_time)
+
+        for dic_mpath in self.mpath_list:
+            if multipath.resume_mpath(dic_mpath["name"]) is False:
+                self.log.info("couldn't resume %s ", dic_mpath["name"])
+                err_mpaths.append(dic_mpath['name'])
+        if err_mpaths:
+            self.fail("error mpaths in suspnd all mpaths: %s" % err_mpaths)
+
+    def remove_add_mpath(self):
```

Comment:
@Naresh-ibm I think this needs to be `test_remove_add_mpath`. Once this is changed, we are ready to merge.